packages/framework/src/Component/Domain/Multidomain/Twig/FilesystemLoader.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Shopsys\FrameworkBundle\Component\Domain\Multidomain\Twig;
+
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Symfony\Bundle\TwigBundle\Loader\FilesystemLoader as BaseFilesystemLoader;
+use Symfony\Component\Config\FileLocatorInterface;
+use Symfony\Component\Templating\TemplateNameParserInterface;
+
+class FilesystemLoader extends BaseFilesystemLoader
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain|null
+     */
+    protected $domain;
+
+    /**
+     * @param \Symfony\Component\Config\FileLocatorInterface $locator
+     * @param \Symfony\Component\Templating\TemplateNameParserInterface $parser
+     * @param string|null $rootPath
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain|null $domain
+     */
+    public function __construct(
+        FileLocatorInterface $locator,
+        TemplateNameParserInterface $parser,
+        ?string $rootPath = null,
+        Domain $domain = null
+    ) {
+        $this->domain = $domain;
+        parent::__construct($locator, $parser, $rootPath);
+    }
+
+    /**
+     * When exists a template with filename.{designId}.html.twig, then it is automatically used
+     * on domain with this {designId} whenever template named filename.html.twig is on input
+     *
+     * @inheritdoc
+     */
+    protected function findTemplate($template, $throw = true)
+    {
+        $templateName = (string)$template;
+        if (strpos($templateName, '@ShopsysShop/Front/') === 0) {
+            $this->assertDomainDependency();
+
+            return $this->findMultidesignTemplate($templateName);
+        }
+
+        return parent::findTemplate($template);
+    }
+
+    protected function assertDomainDependency()
+    {
+        if (!($this->domain instanceof Domain)) {
+            $message = sprintf('Template loader needs an instance of %s class', Domain::class);
+            throw new \Shopsys\FrameworkBundle\Component\Domain\Multidomain\Twig\Exception\MissingDependencyException($message);
+        }
+    }
+
+    /**
+     * @param string $templateName
+     * @return string
+     */
+    protected function findMultidesignTemplate($templateName)
+    {
+        $designId = $this->domain->getDesignId();
+        if ($designId !== null) {
+            $multidesignTemplateName = preg_replace('/^(.*)(\.[^\.]*\.twig)/', '$1.' . $designId . '$2', $templateName);
+            try {
+                return parent::findTemplate($multidesignTemplateName);
+            } catch (\Twig_Error_Loader $loaderException) {
+                if (strpos($loaderException->getMessage(), 'Unable to find template') !== 0) {
+                    $message = sprintf('Unexpected exception when trying to load multidesign template `%s`', $multidesignTemplateName);
+                    throw new \Twig_Error_Loader($message, -1, null, $loaderException);
+                }
+            }
+        }
+
+        return parent::findTemplate($templateName);
```

Comment:
findMultidesignTemplate at the end finds not mutlidesignTemplate :-/ this should not happen in my opinion

Reply:
Hi, what do you suggest? I wanted to name the method "tryToFindMultidesign..." but it seemed too complicated. It is just a fallback, I do not see it as much important here. More on that, it is just a protected helper method...

Reply:
I think this method would be better if it would return MultidesignTemplate|null because of the naming find*
Or if you don't want to refactor it, I would suggest renaming the method

Reply:
ok guys, you convinced me :laughing: :+1: